Bio/SeqIO/QualityIO.py

```diff
@@ -941,18 +941,15 @@ def FastqGeneralIterator(source):
                     raise ValueError("End of file without quality information.")
                 else:
                     raise ValueError("Unexpected end of file")
-            if not seq_string.isalpha():
-                if seq_string:
-                    raise ValueError("Sequence should consists of alphabetical characters")
+            if " " in seq_string or "\t" in seq_string:
+                raise ValueError("Sequence should not contain whitespace.")
+            elif "@" in seq_string:
+                raise ValueError("Sequence should not contain '@'.")
+            seq_len = len(seq_string)
```

Comment:
I'm not entirely sure about relaxing the parser to allow non alphabetical characters... personally I would much prefer an error trying to parse such a file.

Reply:
The commit history is confusing here. I added the `isalpha`, then I removed it again when it turned out that it doesn't do what I thought it does. Check this out:
```python
>>> "あいうえお".isalpha()
True
```
This pull request actually makes the parser _more_ strict, as previously we didn't check for `"@"`.

Reply:
Fair point, is-alpha wasn't as strict as I thought - I mean is-alpha and is-ascii combined? But it did catch the at sign:

```
>>> "@".isalpha()
False
```

Reply:
maq at least seems to allow `.` and `~`:
http://maq.sourceforge.net/fastq.shtml

Reply:
OK, so compared to master, there was no check on the sequence contents at all (other than if a line started with "+").

You are adding a reasonable although new restriction of no "@" in the sequence. I'm OK with that, but would like it much more explicit in the commit history.

The cleanest way to do this is rebase the branch and edit the relevant commit's comment.

The easiest way to do that is we squash-merge the pull request, and write this in the combined commit's message.